addons/account/models/account_payment_term.py

```diff
@@ -48,7 +79,7 @@ def compute(self, company_value, foreign_value, date_ref, currency):
         sign = company_value < 0 and -1 or 1
         result = []
         company_currency = self.env.company.currency_id
-        for line in self.line_ids:
+        for line in self.line_ids.filtered(lambda l: l.value != 'balance') + self.line_ids.filtered(lambda l: l.value == 'balance'):
```

Comment:
`self.line_ids.sorted(lambda line: line.value == 'balance')`? False is before True btw